#! /usr/bin/env python

PACKAGE='controllers'

import roslib;
roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#       	Name		Type      	Level	Description				Default 	Min	Max
gen.add( 	"k_p_xy", 	double_t, 	0, 	"Proportional gain for x and y", 	2.0, 		0, 	10)
gen.add( 	"k_i_xy",	double_t, 	0, 	"Integral gain for x and y", 		0.0, 		0, 	10)
gen.add( 	"k_d_xy",	double_t, 	0, 	"Derivative gain for x and y", 	0.6, 		0, 	10)
gen.add( 	"k_p_z", 	double_t, 	0, 	"Proportional gain for z", 		2.0, 		0, 	10)
gen.add( 	"k_i_z",	double_t, 	0, 	"Integral gain for z", 		0.0, 		0, 	10)
gen.add( 	"k_d_z",	double_t, 	0, 	"Derivative gain for z", 		0.3, 		0, 	10)

#       	Name	Type      	Level	Description		Default 	Min	Max
#gen.add( 	"k_p", 	double_t, 	0, 	"Proportional gain", 	2.0, 		0, 	10) 
#gen.add( 	"k_i",	double_t, 	0, 	"Integral gain", 	0.0, 		0, 	10)
#gen.add( 	"k_d",	double_t, 	0, 	"Derivative gain", 	0.5, 		0, 	10)

exit(gen.generate(PACKAGE, "controllers", "setPID"))
